test/src/main/java/org/apache/accumulo/test/ManagerRepairsDualAssignmentIT.java

```diff
@@ -99,12 +102,21 @@ public void test() throws Exception {
         }
       }
       assertEquals(2, states.size());
+      int deadCount = getZkDeadCount(cluster.getServerContext().getZooReader(),
+          cluster.getServerContext().getInstanceID());
       // Kill a tablet server... we don't care which one... wait for everything to be reassigned
       cluster.killProcess(ServerType.TABLET_SERVER,
           cluster.getProcesses().get(ServerType.TABLET_SERVER).iterator().next());
       Set<TabletMetadata.Location> replStates = new HashSet<>();
       @SuppressWarnings("deprecation")
       TableId repTable = org.apache.accumulo.core.replication.ReplicationTable.ID;
+
+      while (getZkDeadCount(cluster.getServerContext().getZooReader(),
+          cluster.getServerContext().getInstanceID()) <= deadCount) {
+        log.debug("Waiting for dead server to be reported in ZooKeeper. Waiting for next check...");
+        UtilWaitThread.sleep(5000);
+      }
```

Comment:
I think you could use `Wait.waitFor` to watch for this case.

Reply:
changed in 4994f630bf